test/testLauncher/repl.test.js

```diff
@@ -0,0 +1,139 @@
+const assert = require('assert');
+const repl = require('repl');
+const sinon = require('sinon');
+const chokidar = require('chokidar');
+const EventEmitter = require('events');
+
+const texts = require('../../lib/texts');
+const SuitestError = require('../../lib/utils/SuitestError');
+const suitestRepl = require('../../lib/testLauncher/repl');
+
+const sandbox = require('sinon').createSandbox();
+
+// mock repl instance
+class ReplInstance extends EventEmitter {
+	constructor(autoEnd = true) {
+		// console.log('booooo');
+		super();
+		this.context = {};
+		this.close = sinon.stub();
+		this.eval = (a, b, c, f) => f && f();
+		this.defineCommand = sinon.stub();
+
+		if (autoEnd)
+			setInterval(() => this.emitExit(), 10);
+	}
+
+	emitExit() {
+		this.emit('exit');
+	}
+}
+
+describe('repl', () => {
+	beforeEach(() => {
+		process.env.REPL_IPC_PORT = '9999';
+		sandbox.stub(repl, 'start').returns(new ReplInstance());
+	});
+
+	afterEach(() => {
+		sandbox.restore();
+	});
+
+	it('should require IPC port', () => {
+		const replIpcNotAvailable = sandbox.stub(texts, 'replIpcNotAvailable');
+		const port = process.env.REPL_IPC_PORT;
+
+		delete process.env.REPL_IPC_PORT;
+
+		const {startRepl} = suitestRepl;
+		let passed = false;
+
+		try {
+			startRepl();
+		} catch (e) {
+			console.log(replIpcNotAvailable.called);
+			assert.equal(e.code, SuitestError.IPC_ERROR, 'Error code is correct');
+			assert.ok(replIpcNotAvailable.called, 'Error message is correct');
+			passed = true;
+		}
+
+		assert.ok(passed, 'Error was not thrown when it should have been.');
+		process.env.REPL_IPC_PORT = port;
+	});
+
+	it('should handle IPC startup error', async() => {
+		const replIpcErrorInChildProcess = sandbox.stub(texts, 'replIpcErrorInChildProcess');
+
+		suitestRepl.startRepl();
+
+		return new Promise(resolve => {
```

Comment:
why this Promise needed?

Reply:
because one needs to wait until error is thrown in the uncaught exception handler, and returning promise seemed like a good idea to achieve this async test.